### PR TITLE
Cache - new updates by symbol

### PR DESCRIPTION
**Pull request by pcriadoperez**
This fixes an issue that when clearing the updates, they should be cleared for all symbols also.

Report:
(https://discord.com/channels/690203284119617602/690203284727660739/1030802028047376395)

To reproduce:
- Call `npm run cli.js binanceusdm watchOrders --verbose --test`
- Post one market order  `npm run cli.js binanceusdm createOrder ETH/USDT market buy 1 --verbose --test`
- Expect one update, get one update ✅
- Post a second market order `npm run cli.js binanceusdm createOrder ETH/USDT market buy 1 --verbose --test`
- Expect one update to be returned, however `allNewUpdates = 0` and  `this.orders` is returned. ❌

Also looking at this code, I think we should consider in the future rewriting this as I can see there being issues when a user tries to call both watchOrders for a specific symbol and then watchOrders for all symbols. As the limit will be cleared in the first update, affecting the results of the second watchOrders.

### CODE REVIEW

**Comment by carlosmiei** [comment]
@pcriadoperez thanks, finally the issue was found!

btw do you think it's feasible to add a test against this issue?

**Comment by pcriadoperez** [comment]
Great point! However,  as I mentioned I think even though this fix

> @pcriadoperez thanks, finally the issue was found!
> 
> btw do you think it's feasible to add a test against this issue?

Great point! I'll add one.

I'm also thinking for the case when a user is using both `watchOrders` for a specific symbol and all symbols at the same time. What should be the wanted behavior of newUpdates?
```javascript
 // User runs
 await watchOrders(BTC/USDT)
 await watchOrders()
 ```
 Option A - watchOrders returns all order updates since last call of watchOrders 
 Option B - watchOrders returns all order updates without the updates returned by BTC/USDT
 
To me option A makes more sense, but I can see you could discuss option B is also correct as the updates for the orders of BTC/USDT aren't really new as they have already been returned in the first call.

**Comment by carlosmiei** [comment]
> 



> Great point! However, as I mentioned I think even though this fix
> 
> > @pcriadoperez thanks, finally the issue was found!
> > btw do you think it's feasible to add a test against this issue?
> 
> Great point! I'll add one.
> 
> I'm also thinking for the case when a user is using both `watchOrders` for a specific symbol and all symbols at the same time. What should be the wanted behavior of newUpdates?
> 
> ```js
>  // User runs
>  await watchOrders(BTC/USDT)
>  await watchOrders()
> ```
> 
> Option A - watchOrders returns all order updates since last call of watchOrders Option B - watchOrders returns all order updates without the updates returned by BTC/USDT
> 
> To me option A makes more sense, but I can see you could discuss option B is also correct as the updates for the orders of BTC/USDT aren't really new as they have already been returned in the first call.

@pcriadoperez actually I would say that A is the only correct option here 😅,  IMO both promises should be "independent" and don't interfere with the result of the other

**Comment by frosty00** [comment]
looks correct to me

File: python/ccxt/pro/base/cache.py [hunk 1]
```diff
@@ -77,12 +77,14 @@ def getLimit(self, symbol, limit):
...
         self._deque.append(item)
-        if self._clear_updates_by_symbol.get(item['symbol']):
-            self._clear_updates_by_symbol[item['symbol']] = False
-            self._new_updates_by_symbol[item['symbol']] = 0
         if self._clear_all_updates:
             self._clear_all_updates = False
+            self._clear_updates_by_symbol = {}
             self._all_new_updates = 0
+            self._new_updates_by_symfbol = {}
+        if self._clear_updates_by_symbol.get(item['symbol']):
```
Comment (carlosmiei) [inline python/ccxt/pro/base/cache.py:85]: @pcriadoperez build is broken, can you check it?

**Comment by frosty00** [comment]
i can't reproduce on spot markets... hmm

**Comment by frosty00** [comment]
cannot reproduce this issue at all, i suggest to avoid merging for now

**Comment by frosty00** [comment]
finally i am able to reproduce, it seems only possible in python due to the nature of fastclient

**Comment by frosty00** [comment]
ok am able to reproduce this, will merge

**Comment by frosty00** [comment]
> Also looking at this code, I think we should consider in the future rewriting this as I can see there being issues when a user tries to call both watchOrders for a specific symbol and then watchOrders for all symbols. As the limit will be cleared in the first update, affecting the results of the second watchOrders.

Actually it was designed with this in mind, and it's pretty difficult to come up with a better design than the current one imo, it took multiple attempts to come up with this design and I think it is quite good because it naturally buffers updates on the exchange.

**Comment by pcriadoperez** [comment]
> > Also looking at this code, I think we should consider in the future rewriting this as I can see there being issues when a user tries to call both watchOrders for a specific symbol and then watchOrders for all symbols. As the limit will be cleared in the first update, affecting the results of the second watchOrders.
> 
> Actually it was designed with this in mind, and it's pretty difficult to come up with a better design than the current one imo, it took multiple attempts to come up with this design and I think it is quite good because it naturally buffers updates on the exchange.

@frosty00 you are right. After writing some of the tests I saw that the promises do work independently as we wanted. So I think we are good for now.

**Pull request merged**